lntest/itest/lnd_funding_test.go

```diff
@@ -970,3 +976,190 @@ func testBatchChanFunding(net *lntest.NetworkHarness, t *harnessTest) {
+	}
+
+	pendingUpdate, err := net.OpenPendingChannel(alice, bob, openReq)
+	require.NoError(t.t, err)
+
+	if !zeroConf {
+		assertNumOpenChannelsPending(t, alice, bob, 1)
+	}
+
+	// Remove the ChannelAcceptor.
+	cancel()
+
+	// Wait for the funding transaction to hit the mempool.
+	_, err = waitForTxInMempool(net.Miner.Client, minerMempoolTimeout)
+	require.NoError(t.t, err)
+
+	fundingTxHash, _ := chainhash.NewHash(pendingUpdate.Txid)
+	fundingTx, err := net.Miner.Client.GetRawTransaction(fundingTxHash)
+	require.NoError(t.t, err)
+
+	// Fetch the input so it can be spent by a different transaction.
+	inputHash := &fundingTx.MsgTx().TxIn[0].PreviousOutPoint.Hash
+	inputTx, err := net.Miner.Client.GetRawTransaction(inputHash)
+	require.NoError(t.t, err)
+
+	// Create the conflict transaction.
+	conflictTx := wire.NewMsgTx(2)
+	conflictTx.AddTxIn(fundingTx.MsgTx().TxIn[0])
+
+	// Create the output the conflict address will pay out to.
+	addrReq := &lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_WITNESS_PUBKEY_HASH,
+	}
+
+	addrResp, err := alice.NewAddress(ctxb, addrReq)
+	require.NoError(t.t, err)
+
+	addr, err := btcutil.DecodeAddress(
+		addrResp.Address, net.Miner.ActiveNet,
+	)
+	require.NoError(t.t, err)
+
+	addrScript, err := txscript.PayToAddrScript(addr)
+	require.NoError(t.t, err)
+
+	output := &wire.TxOut{
+		PkScript: addrScript,
+		Value:    int64(2_000),
+	}
+	conflictTx.AddTxOut(output)
+
+	var conflictTxBytes bytes.Buffer
+	err = conflictTx.Serialize(&conflictTxBytes)
+	require.NoError(t.t, err)
+
+	// Sign the conflict transaction.
+	signDesc := signrpc.SignDescriptor{
+		Output: &signrpc.TxOut{
+			Value:    inputTx.MsgTx().TxOut[0].Value,
+			PkScript: inputTx.MsgTx().TxOut[0].PkScript,
+		},
+		InputIndex: 0,
+	}
+
+	signResp, err := alice.SignerClient.ComputeInputScript(
+		ctxb, &signrpc.SignReq{
+			SignDescs:  []*signrpc.SignDescriptor{&signDesc},
+			RawTxBytes: conflictTxBytes.Bytes(),
+		},
+	)
+	require.NoError(t.t, err)
+
+	// Populate the witness and confirm the transaction.
+	conflictTx.TxIn[0].Witness = signResp.InputScripts[0].Witness
+
+	_, err = net.Miner.GenerateAndSubmitBlock(
+		[]*btcutil.Tx{btcutil.NewTx(conflictTx)}, -1, time.Time{},
+	)
+	require.NoError(t.t, err)
+
+	// Alice should no longer see the channel as pending.
+	err = wait.NoError(func() error {
+		aliceNumChans, err := numOpenChannelsPending(ctxb, alice)
+		if err != nil {
+			return err
+		}
+
+		if aliceNumChans != 0 {
+			return fmt.Errorf("expected alice to have no pending " +
+				"channels")
+		}
+
+		return nil
+	}, defaultTimeout)
+	require.NoError(t.t, err)
+
+	// Alice's ListChannels output should be empty.
+	err = wait.NoError(func() error {
+		req := &lnrpc.ListChannelsRequest{}
+		resp, err := alice.ListChannels(ctxb, req)
+		if err != nil {
+			return err
+		}
+
+		if len(resp.Channels) != 0 {
+			return fmt.Errorf("expected alice to have no channels")
+		}
+
+		return nil
+	}, defaultTimeout)
```

Comment:
Can we also test that Alice is able to respend any inputs that should be unlocked after the pending channel is cancelled?